packages/grid/x-data-grid/src/tests/sorting.DataGrid.test.tsx

```diff
@@ -450,5 +453,51 @@ describe('<DataGrid /> - Sorting', () => {
       fireEvent.click(screen.getAllByRole('columnheader')[0]);
       expect(getColumnValues(0)).to.deep.equal(['Puma', 'Nike', 'Adidas']);
     });
+
+    it('should not allow to initialize the sorting with several items', () => {
+      expect(() => {
+        render(
+          <Test
+            columns={[{ field: 'id', type: 'number' }, { field: 'brand' }]}
+            rows={[
+              {
+                id: 0,
+                brand: 'Nike',
+              },
+              {
+                id: 1,
+                brand: 'Nike',
+              },
+              {
+                id: 2,
+                brand: 'Adidas',
+              },
+              {
+                id: 3,
+                brand: 'Puma',
+              },
+            ]}
+            initialState={{
+              sorting: {
+                sortModel: [
+                  {
+                    field: 'brand',
+                    sort: 'asc',
+                  },
+                  {
+                    field: 'id',
+                    sort: 'desc',
+                  },
+                ],
+              },
+            }}
+          />,
+        );
+
+        expect(getColumnValues(0)).to.deep.equal(['2', '0', '1', '3']);
+      }).toWarnDev(
```

Comment:
Why do we wanter `console.error` + `.toErrorDev` instead of `console.warn` + `.toWarnDev` ?

I'm making the filtering test pass by removing the proptypes check.

Reply:
`console.error` is more noticeable than a warning. Also, users could have disabled warnings to reduce noise in the DevTools console but kept errors on to get React errors.

<img width="300" src="https://user-images.githubusercontent.com/42154031/153862000-2a3aea9c-3008-4ccb-ac08-cea10ca75ed4.png" alt="image" style="max-width: 100%;">